src/sysctl/sysctl.c

```diff
@@ -26,25 +26,71 @@ static char **arg_prefixes = NULL;
 static bool arg_cat_config = false;
 static bool arg_no_pager = false;
 
+typedef struct Option {
+        char *key;
+        char *value;
+        bool ignore_failure;
+} Option;
+
+static Option *option_free(Option *o) {
+        if (!o)
+                return NULL;
+
+        free(o->key);
+        free(o->value);
+
+        return mfree(o);
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(Option*, option_free);
+DEFINE_HASH_OPS_WITH_VALUE_DESTRUCTOR(option_hash_ops, char, string_hash_func, string_compare_func, Option, option_free);
```

Comment:
Oh well, looks like we don't even have the `HASH_OPS` destructor helpers in RHEL 8 (they were introduced in v240 by https://github.com/systemd/systemd/pull/10920). Given that's a quite large backport, I wonder what should be the next steps, since the `-` prefix is necessary there to make stuff work in certain containers (see https://github.com/systemd/systemd/issues/13177#issuecomment-514931461 for reference).

/cc @dtardon @msekletar